File: 配套资源/配套代码/程序设计竞赛专题挑战教程-代码/第4章错误票据.py
#错误票据 lanqiaoOJ 205

n = int(input())
a = []
for i in range(n):
    num = input().split()
    for j in range(len(num)):
        a.append(int(num[j]))    #读取n行数据，存到a[]
a.sort()
for i in range(a[0],a[0]+len(a)):
    if i not in a:     ans1 = i
    if a.count(i)==2:  ans2 = i
print(ans1,ans2)

File: 配套资源/配套代码/程序设计竞赛专题挑战教程-代码/第4章错误票据.cpp
//错误票据 lanqiaoOJ 205 
#include<bits/stdc++.h>
using namespace std;
const int N = 1e4+10;
int a[N];
int main(){
    int n; cin >> n;
    int cnt = 0;
    while(scanf("%d", &a[cnt]) != EOF)   cnt++;  //注意读数据的写法
    sort(a, a+cnt);
    int ans1, ans2;
    for(int i = 1; i < cnt; i++) {
        if(a[i] - a[i-1] > 1)   ans1 = a[i-1]+1;
        if(a[i] == a[i-1])      ans2 = a[i];
    }
    cout << ans1 << ' ' << ans2;
    return 0;
}
